This produces a working (tested) image.

mbacchi opened the pull request:
I think the main issue was that the image was being written as qcow2,
and this forces a raw image to be output.  Also added some extra packages
that needed to be installed, and added jenkins user to sudoers file.

Finally added a very brief readme with instructions on what is required
to get this to build on a centos 7 machine.

mbacchi committed:
This produces a working (tested) image.

I think the main issue was that the image was being written as qcow2,
and this forces a raw image to be output.  Also added some extra packages
that needed to be installed, and added jenkins user to sudoers file.

Finally added a very brief readme with instructions on what is required
to get this to build on a centos 7 machine.

gholms commented on Makefile.in:
```diff
@@ -20,6 +20,9 @@ build: euimage
 	mkdir -p build/image
+	sed -i.bak 's/^#user = "root"/user = "root"/' /etc/libvirt/qemu.conf
+	sed -i.bak 's/^#group = "root"/group = "root"/' /etc/libvirt/qemu.conf
+	systemctl restart libvirtd.service
```
Yeah, this is going to prevent make from running as anyone other than root.  At most this step, along with an explanation of what _specifically_ it fixes and why, should go in the README file.

gholms commented on README.md:
```diff
@@ -0,0 +1,5 @@
+
+### On a CentOS 7 system:
+
+    yum install libguestfs-tools autoconf virt-install virsh virt-sysprep virt-sparsify euca2ools python-requestbuilder
```
euca2ools should pull in python-requestbuilder on its own.  Did you have to install it separately?

gholms commented on centos-7-builder.ks.in:
```diff
@@ -5,7 +5,8 @@ keyboard us
 network --bootproto=dhcp --device=eth0 --onboot=on
-rootpw --iscrypted $6$t5nrfmzb2CiWcTA7$12PVlif6C2cS8pEFhVAlFqOewvj0UlCe0czyOEeVzIrpc7UEqu0kFITHgK9pL.xjxPURA4YCqmkagFa6E6IEt0
+authconfig --enableshadow --passalgo=sha512
+rootpw --iscrypted $6$OeV7FnQ9.T.pecld$Rab/mXpQRe0kKW3m3L1uPVZWvRZezT4j8mMtn7wgJp/eZlpsroKEz8LRYTHX6Yg5hl90yMYZ4lcoUwXYcf8R10
```
The password was randomized, so let's leave this change out.

gholms commented:
I'm fairly sure this image doesn't need access to any NFS shares, but I could be forgetting something.  Can you think of what it might be?  If not, let's throw out nfs-utils and /srv/release.

What is using wget?

mbacchi committed:
updated kickstart.in with packages, cloud-init parameters

removed qemu.conf modifications from Makefile.in but added info
to README for instructions on how to make such changes if necessary

this image won't be able to grow the root filesystem, I'm still researching
how to get that to work on eucalyptus or AWS reliably since cloud-init
doesn't explicitly say they support this yet in version 0.7.7
